src/select.coffee

```diff
@@ -346,11 +371,26 @@ Ember.AddeparMixins.ResizeHandlerMixin,
     value
   .property 'selectableOptions.[]', 'highlightedIndex'
 
+  setFocus: ->
+    activeElement = $(document.activeElement)[0]
+    selectComponent = @$()[0]
+    if selectComponent.contains(activeElement) or selectComponent is activeElement
+      @set 'hasFocus', yes
+    else
+      @set 'hasFocus', no
+
   bodyClick: -> @send 'hideDropdown'
 
   keyDown: (event) ->
-    # show dropdown if dropdown is not already showing
-    return @set('showDropdown', yes) unless @get 'showDropdown'
+    return if @get('isDestroyed') or @get('isDestroying')
+    @setFocus()
+    # show dropdown if it is not already showing
+    # and the keycode should be in the list of accepted keys to show dropdown
+    # [Spacebar, Enter, Up, Down, 'A'..'Z','a..z','0..9']
+    acceptedKeys = @get 'acceptedKeys'
+    if acceptedKeys[event.keyCode] and not @get 'showDropdown'
+      return @set('showDropdown', yes)
```

Comment:
the result of a `@set` isn't very interesting; I think you should `set` and then `return` on a new line.